pkg/plugin/work-in-progress/event_handler.go

```diff
@@ -178,3 +146,89 @@ func (gh *GitHubWIPPRHandler) checkTitleAndSetStatus(log log.Logger, pullRequest
 	return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
 
 }
+
+func (gh *GitHubWIPPRHandler) updateTitleAndResetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
+	change := gh.getRepositoryChange(pullRequest)
+	statusService := gh.getStatusService(log, change)
+	labels := gh.getPullRequestLabels(log, change, pullRequest)
+	configuration := LoadConfiguration(log, change)
+
+	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
+	if !labelExists {
+		if prefix, ok := gh.IsWorkInProgress(*pullRequest.Title, configuration); ok {
+			*pullRequest.Title = strings.TrimPrefix(*pullRequest.Title, prefix)
+			err := gh.Client.EditPullRequest(change, *pullRequest.Number, pullRequest)
+			if err != nil {
+				log.Warnf("failed to update PR title [%q]. cause: %s", *pullRequest, err)
+			}
+		}
+		return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
+	}
+	return statusService.Failure(InProgressMessage, InProgressDetailsPageName)
+}
+
+func (gh *GitHubWIPPRHandler) checkLabelAndSetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
+	change := gh.getRepositoryChange(pullRequest)
+	statusService := gh.getStatusService(log, change)
+	labels := gh.getPullRequestLabels(log, change, pullRequest)
+	configuration := LoadConfiguration(log, change)
+
+	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
+	if labelExists {
+		return statusService.Failure(InProgressMessage, InProgressDetailsPageName)
+	}
+	return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
+}
+
+func (gh *GitHubWIPPRHandler) getPullRequestLabels(log log.Logger, change scm.RepositoryChange, pullRequest *gogh.PullRequest) []*gogh.Label {
+	labels, err := gh.Client.ListPullRequestLabels(change, *pullRequest.Number)
+	if err != nil {
+		log.Warnf("failed to list labels on PR [%q]. cause: %s", *pullRequest, err)
+	}
+	return labels
+}
+
+func (gh *GitHubWIPPRHandler) hasWorkInProgressLabel(labels []*gogh.Label, wipLabel string) bool {
+	for _, label := range labels {
+		if label.GetName() == wipLabel {
+			return true
+		}
+	}
+	return false
+}
+
+// IsWorkInProgress checks if title is marked as Work In Progress
+func (gh *GitHubWIPPRHandler) IsWorkInProgress(title string, config PluginConfiguration) (string, bool) {
+	prefixes := defaultPrefixes
+	if len(config.Prefix) != 0 {
+		if config.Combine {
+			prefixes = append(prefixes, config.Prefix...)
+		} else {
+			prefixes = config.Prefix
+		}
+	}
+	return gh.hasPrefix(strings.ToLower(title), prefixes)
+}
+
+func (gh *GitHubWIPPRHandler) hasPrefix(title string, prefixes []string) (string, bool) {
+	for _, prefix := range prefixes {
+		pattern := `(?mi)^(\[|\()?` + prefix + `(\]|\))?(:| )+`
+		if match, _ := regexp.MatchString(pattern, title); match {
+			return prefix, true
+		}
+	}
+	return "", false
+}
+
+func (gh *GitHubWIPPRHandler) getRepositoryChange(pullRequest *gogh.PullRequest) scm.RepositoryChange {
```

Comment:
This should be placed next to the `scm.RepositoryChange` struct as its constructor. And by convention, the name should start with "New"

Reply:
there is already such a function: https://github.com/arquillian/ike-prow-plugins/blob/master/pkg/github/service/repository_change.go so please use this one